refactor(linter): move finishing default diagnostic message to `GraphicalReporter`

Sysix opened the pull request:
Now every lint output is owned by is right OutputFormatter and his DiagnosticReporter 🥳 
Next step is to setup a snapshot Tester, so I can remove the ToDos.

Reorded some lines so the outfor is now for: `cargo run -p oxlint -- test.js --max-warnings=2`
```
Found 4 warnings and 0 errors.
Exceeded maximum number of warnings. Found 4.
Finished in 5ms on 1 file with 97 rules using 24 threads.
```

and for `cargo run -p oxlint -- test.js`

```
Found 4 warnings and 0 errors.
Finished in 5ms on 1 file with 97 rules using 24 threads.
```

The output time and warnings/error count wil be always printed.

Sysix commented:
* **#8683** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8683?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8683?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=8683).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented on apps/oxlint/src/lint.rs:
```diff
@@ -306,6 +311,15 @@ impl LintRunner {
     }
+
+    fn check_for_writer_error(error: std::io::Error) -> Result<(), std::io::Error> {
+        // Do not panic when the process is skill (e.g. piping into `less`).
```
I had a typo here `skill` -> killed.

Boshen commented on apps/oxlint/src/output_formatter/checkstyle.rs:
```diff
@@ -17,6 +17,10 @@ impl InternalFormatter for CheckStyleOutputFormatter {
         writeln!(writer, "flag --rules with flag --format=checkstyle is not allowed").unwrap();
     }
 
+    fn lint_command_info(&self, _: &super::LintCommandInfo) -> Option<String> {
```
You can make the trait default implementation return `None` so you don't need to add these blanket implementations.

Boshen commented on apps/oxlint/src/result.rs:
```diff
@@ -17,22 +16,14 @@ pub enum CliRunResult {
-    /// Whether or not to print a summary of the results
-    pub print_summary: bool,
+    /// The exit unix code for, in general 0 or 1 (from `--deny-warnings` or `--max-warnings` for example)
+    pub exit_code: u8,
```
https://doc.rust-lang.org/beta/std/process/struct.ExitCode.html would be nice.

Sysix commented:
implemented the requested changes :)

Boshen commented:
### Merge activity

* **Jan 24, 11:38 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
refactor(linter): move finishing default diagnostic message to `GraphicalReporter` (#8683)

Now every lint output is owned by is right OutputFormatter and his DiagnosticReporter 🥳
Next step is to setup a snapshot Tester, so I can remove the ToDos.

Reorded some lines so the outfor is now for: `cargo run -p oxlint -- test.js --max-warnings=2`
```
Found 4 warnings and 0 errors.
Exceeded maximum number of warnings. Found 4.
Finished in 5ms on 1 file with 97 rules using 24 threads.
```

and for `cargo run -p oxlint -- test.js`

```
Found 4 warnings and 0 errors.
Finished in 5ms on 1 file with 97 rules using 24 threads.
```

The output time and warnings/error count wil be always printed.
